packages/web-components/src/components/va-date/va-date.tsx

```diff
@@ -99,9 +116,10 @@ export class VaDate {
       year > maxYear ||
       month < minMonths ||
       month > maxMonths ||
-      day < minMonths ||
-      day > daysForSelectedMonth.length ||
-      !day ||
+      (!this.monthYearOnly && (
+        day < minMonths ||
+        day > daysForSelectedMonth.length ||
+        !day)) ||
```

Comment:
This is so that we only do the `day` boolean comparisons if we're a regular date component, i.e. not the `monthYearOnly` variant.